aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/Assertions.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.sdk.kotlin.crt
+
+import kotlinx.cinterop.ExperimentalForeignApi
+import libcrt.AWS_OP_SUCCESS
+
+/**
+ * Assert that this [Int] return value is `AWS_OP_SUCCESS`. If not, throw [CrtRuntimeException] with the given evaluated
+ * [lazyMessage].
+ */
+@OptIn(ExperimentalForeignApi::class)
+internal inline fun Int.awsAssertOpSuccess(lazyMessage: () -> String) {
```

Comment:
**style**: having this be a parameter rather than an extension better matches the stdlib `check`/`require` functions. It also is more readable in my opinion (but that's just like my opinion of course)